googletest/include/gtest/gtest.h

```diff
@@ -2116,9 +2116,14 @@ GTEST_API_ AssertionResult DoubleLE(const char* expr1, const char* expr2,
 // of the dummy variable name, thus allowing multiple SCOPED_TRACE()s
 // to appear in the same block - as long as they are on different
 // lines.
-#define SCOPED_TRACE(message) \
+#define SCOPED_TRACE(...) GTEST_VA_SELECT_(SCOPED_TRACE, __VA_ARGS__) 
+#define SCOPED_TRACE_1(message) \
```

Comment:
It can still do that.
Does the __LINE__ that forms the variable name have to be the same as the
'line' that's printed in the SCOPED_TRACE?


On Tue, Nov 24, 2015 at 1:48 PM, ⭐caitp⭐ <notifications@github.com> wrote:

> In googletest/include/gtest/gtest.h
> <https://github.com/google/googletest/pull/620#discussion_r45776786>:
>
> > @@ -2116,9 +2116,14 @@ GTEST_API_ AssertionResult DoubleLE(const char* expr1, const char* expr2,
> >  // of the dummy variable name, thus allowing multiple SCOPED_TRACE()s
> >  // to appear in the same block - as long as they are on different
> >  // lines.
> > -#define SCOPED_TRACE(message) \
> > +#define SCOPED_TRACE(...) GTEST_VA_SELECT_(SCOPED_TRACE, __VA_ARGS__)
> > +#define SCOPED_TRACE_1(message) \
>
> The macro is declaring a variable in the calling scope, using a name
> derived from the line of the call. So I'm not positive that can fully work
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/googletest/pull/620/files#r45776786>.
>